### DISCUSSION TITLE
Running PyTorch on Apple Silicon

### DISCUSSION THREAD

**Discussion by msp99000** [mrdbourke/pytorch-deep-learning#255] [Q&A] [answered]
Slightly off topic, was wondering if there's anyone who's running PyTorch on M1/M2 Mac. I have a M2 Mac and I did not quite get how to run GPU enabled PyTorch

**Comment by AlienSarlak** [comment, marked as answer]
Hi,

Unfortunately, I don't have an Apple system but have a look at the following:

https://towardsdatascience.com/installing-pytorch-on-apple-m1-chip-with-gpu-acceleration-3351dc44d67c

I hope it can help you.

**Reply by msp99000** [reply]
thanks, it's simple with conda

**Comment by shmuhammadd** [comment]
Hy,

I am running M1. 

This [setup video](https://www.youtube.com/watch?v=Zx2MHdRgAIc) by @mrdbourke could be helpful.

**Reply by msp99000** [reply]
thanks a lot man

**Comment by mrdbourke** [comment]
Hi @msp99000,

@shmuhammad2004 and @AlienSarlak are right there are a couple of methods to get it setup.

Try this code snippet as well to use the Apple Silicon GPU from the PyTorch Cheatsheet - https://www.learnpytorch.io/pytorch_cheatsheet/#device-agnostic-code-using-pytorch-on-cpu-gpu-or-mps

```python
# Setup device-agnostic code 
if torch.cuda.is_available():
    device = "cuda" # NVIDIA GPU
elif torch.backends.mps.is_available():
    device = "mps" # Apple GPU
else:
    device = "cpu" # Defaults to CPU if NVIDIA GPU/Apple GPU aren't available

print(f"Using device: {device}")
```

**Reply by msp99000** [reply]
Hi @mrdbourke , thanks I followed the steps and installed pytorch in conda environment though my Jupyter Notebook doesn't recognise it because I have Jupyter Lab installed via pip3. I've read the article on how to troubleshoot this. Thanks again
